flang/test/Lower/OpenMP/omp-flush.f90

```diff
@@ -0,0 +1,78 @@
+! This test checks lowering of OpenMP Flush Directive.
```

Comment:
Can you add flush with a list as well?

Reply:
That can't be done here, since semantic checks for the clauses of Flush construct are missing. They won't pass the semantic phase.
I've marked that as a TODO.
Further more kiran's patch in MLIR side of things tests those scenarios i.e `omp.flush(%arg0, %arg0: !llvm.i32, !llvm.i32)`

Reply:
What is missing exactly? The `flush` directive has information in the `OMP.td` file.

Reply:
`flush` has memory order clauses, are semantics done for them ?

Reply:
The semantic side seems ok. The parser should use `OmpClause` instead of specific set le memory order clauses.

Reply:
Is this okay ?
```
!$OMP FLUSH(a,b,c)
        c = a + b

parse-tree dump :

ExecutionPartConstruct -> ExecutableConstruct -> OpenMPConstruct -> OpenMPStandaloneConstruct -> OpenMPFlushConstruct
| | | Verbatim
| | | OmpObjectList -> OmpObject -> Designator -> DataRef -> Name = 'a'
| | | OmpObject -> Designator -> DataRef -> Name = 'b'
| | | OmpObject -> Designator -> DataRef -> Name = 'c'

```

Reply:
@SouraVX Variable list in flush is supported right? Otherwise how will the following test pass?
https://github.com/llvm/llvm-project/blob/a73166a45204378f6f8b4a6ff2d962f0ff56d51e/flang/test/Semantics/omp-clause-validity01.f90#L463

Reply:
I'll revise soon to make appropriate changes.